sql/src/main/java/io/crate/operation/projectors/IndexWriterProjector.java

```diff
@@ -49,8 +50,34 @@
 
 public class IndexWriterProjector extends AbstractIndexWriterProjector {
 
-    private final BytesRefGenerator generator;
-    private final int sourceColumnIndex;
+    private final SourceInjectorRow row;
+
+    private static final class SourceInjectorRow implements Row{
+
+        private final int idx;
+        private final BytesRefGenerator generator;
+        Row delegate = null;
+
+        private SourceInjectorRow(int idx, BytesRefGenerator generator) {
+            this.idx = idx;
+            this.generator = generator;
+        }
+
+
+        @Override
+        public int size() {
+            return delegate.size();
```

Comment:
this is a private class, which is only used in this projector. the main reason is to set the delegate for this particular case, so i think this should be fine